arceos-vmm/build.rs

```diff
@@ -75,11 +77,118 @@ fn open_output_file() -> fs::File {
+        if let Some(guest_config) = config_path.first() {
+            let config =
+                read_toml_file(guest_config.to_str().expect("Path contains invalid UTF-8"))
+                    .expect("failed to read config file");
+            if let Some(image_location) = config.get("image_location") {
+                let location: &str = image_location.as_str().unwrap();
+                if location == "memory" {
+                    let kernel_path = convert_to_absolute(
+                        "configs",
+                        config.get("kernel_path").unwrap().as_str().unwrap(),
+                    );
+
+                    // If have dtb_path, include it.
+                    writeln!(
+                        out_file,
+                        r#"pub fn load_dtb() -> Option<&'static [u8]> {{ "#
+                    )?;
+                    if let Some(dtb_path) = config.get("dtb_path") {
+                        let dtb_path = convert_to_absolute("configs", dtb_path.as_str().unwrap());
+                        // use include_bytes! load image
+                        writeln!(out_file, r#"    Some(include_bytes!({:?}))"#, dtb_path)?;
+                    } else {
+                        writeln!(out_file, r#"    None"#)?;
+                    };
+
+                    // If have bios_path, include it.
+                    writeln!(
+                        out_file,
+                        r#"pub fn load_bios() -> Option<&'static [u8]> {{ "#
+                    )?;
+                    if let Some(bios_path) = config.get("bios_path") {
+                        let bios_path = convert_to_absolute("configs", bios_path.as_str().unwrap());
+                        // use include_bytes! load image
+                        writeln!(out_file, r#"    Some(include_bytes!({:?}))"#, bios_path)?;
+                    } else {
+                        writeln!(out_file, r#"    None"#)?;
+                    };
+
+                    writeln!(out_file, r#"}}"#)?;
+
+                    writeln!(
+                        out_file,
+                        r#"pub fn load_kernel() -> Option<&'static [u8]> {{ "#
+                    )?;
+                    // use include_bytes! load image
+                    writeln!(out_file, r#"    Some(include_bytes!({:?}))"#, kernel_path)?;
+                    writeln!(out_file, r#"}}"#)?;
+
+                    return Ok(());
+                }
+            }
+        }
+    }
+    writeln!(
+        out_file,
+        r#"
+pub fn load_dtb() -> Option<&'static [u8]> {{ 
+    None
+}}
+    
+pub fn load_kernel() -> Option<&'static [u8]> {{ 
+    None
+}}
+
+pub fn load_bios() -> Option<&'static [u8]> {{ 
+    None
+}}
+"#
+    )?;
+    Ok(())
+}
+
 fn main() -> io::Result<()> {
+    let platform = env::var("AX_PLATFORM").unwrap_or("".to_string());
+    let platform_family = env::var("AX_PLATFORM").unwrap_or("".to_string());
+    let config_toml_path = get_config_paths();
```

Comment:
I think this function is duplicated with the already existed `get_configs` function right?

Reply:
resolve this please